lib/strings.nix

```
@@ -1338,4 +1336,15 @@ rec {
         else if k == 2 then infixDifferAtMost2 ainfix binfix
         else levenshtein ainfix binfix <= k;
       in f;
-}
+} // (
+  let
+    head_tail_warn_msg = name:
+      ''
+        lib.strings.${name} only supports lists; use lib.lists.${name} instead.
+        The lib.strings.${name} alias is deprecated and will be removed after the NixOS 2024.11 release.
+      '';
+  in {
+    # warnings added 2024-05-21
+    head = x: lib.warn (head_tail_warn_msg "head") (builtins.head x);
+    tail = x: lib.warn (head_tail_warn_msg "tail") (builtins.tail x);
```

```suggestion
    head = lib.warn (head_tail_warn_msg "head") builtins.head;
    tail = lib.warn (head_tail_warn_msg "tail") builtins.tail;
```

This way the error gets thrown only once (generally how deprecations are done in `lib`)